neo/rawio/biocamrawio.py

```diff
@@ -0,0 +1,203 @@
+        # transform to slice if possible
+        if sorted(channel_indexes) == channel_indexes and np.all(np.diff(channel_indexes) == 1):
+            channel_ids = slice(channel_indexes[0], channel_indexes[0] + len(channel_indexes))
+        return data[:, channel_indexes]
+
+
+def openBiocamFile(filename, mea_pitch, verbose=False):
+    """Open a Biocam hdf5 file, read and return the recording info, pick te correct method to access raw data,
+    and return this to the caller."""
+    try:
+        import h5py
+        HAVE_H5PY = True
+    except ImportError:
+        HAVE_H5PY = False
+    assert HAVE_H5PY, 'h5py is not installed'
+
+    rf = h5py.File(filename, 'r')
+    # Read recording variables
+    recVars = rf.require_group('3BRecInfo/3BRecVars/')
+    # bitDepth = recVars['BitDepth'].value[0]
+    # maxV = recVars['MaxVolt'].value[0]
+    # minV = recVars['MinVolt'].value[0]
+    nFrames = recVars['NRecFrames'][0]
+    samplingRate = recVars['SamplingRate'][0]
+    signalInv = recVars['SignalInversion'][0]
+    # Read chip variables
+    chipVars = rf.require_group('3BRecInfo/3BMeaChip/')
+    nCols = chipVars['NCols'][0]
+    # Get the actual number of channels used in the recording
+    file_format = rf['3BData'].attrs.get('Version')
+    if file_format == 100:
+        nRecCh = len(rf['3BData/Raw'][0])
+    elif (file_format == 101) or (file_format == 102):
+        nRecCh = int(1. * rf['3BData/Raw'].shape[0] / nFrames)
+    else:
+        raise Exception('Unknown data file format.')
+
+    if verbose:
+        print('# 3Brain data format:', file_format, 'signal inversion', signalInv)
+        print('#       signal range: ', recVars['MinVolt'][0], '- ', recVars['MaxVolt'][0])
+        print('# channels: ', nRecCh)
+        print('# frames: ', nFrames)
+        print('# sampling rate: ', samplingRate)
+    # get channel locations
+    r = (rf['3BRecInfo/3BMeaStreams/Raw/Chs'][()]['Row'] - 1) * mea_pitch
+    c = (rf['3BRecInfo/3BMeaStreams/Raw/Chs'][()]['Col'] - 1) * mea_pitch
+    rawIndices = np.vstack((r, c)).T
+    # assign channel numbers
+    chIndices = np.array([(x - 1) + (y - 1) * nCols for (y, x) in rawIndices])
+    # determine correct function to read data
+    if verbose:
+        print("# Signal inversion is " + str(signalInv) + ".")
+        print("# If your spike sorting results look wrong, invert the signal.")
+    if (file_format == 100) & (signalInv == 1):
+        read_function = readHDF5t_100
+    elif (file_format == 100) & (signalInv == -1):
+        read_function = readHDF5t_100_i
+    if ((file_format == 101) | (file_format == 102)) & (signalInv == 1):
+        read_function = readHDF5t_101
+    elif ((file_format == 101) | (file_format == 102)) & (signalInv == -1):
+        read_function = readHDF5t_101_i
+    else:
+        raise RuntimeError("File format unknown.")
+    return rf, nFrames, samplingRate, nRecCh, chIndices, file_format, signalInv, rawIndices, read_function
```

Comment:
I would use a dict as output.
So we can enhance it more easily.

Reply:
done